apps/explorer/src/ui/StatAmount.tsx

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { SUI_TYPE_ARG, Coin } from '@mysten/sui.js';
+
+import { numberSuffix } from '../utils/numberUtil';
+
+import { CoinFormat, formatBalance } from '~/hooks/useFormatCoin';
+import { Heading } from '~/ui/Heading';
+import { Text } from '~/ui/Text';
+
+export interface StatAmountProps {
+    amount: bigint | string | number;
+    currency?: string;
+    dollarAmount?: number;
+    date?: string;
```

Comment:
We also might want to think about taking a date object or time stamp and doing the formatting instead of just a formatted string.

Reply:
I think it depends on how consistent we expect this display to be.

Reply:
I think keeping this component clean is the right approach. Not sure if we want to enforce the date format. Using as a string makes the most sense

Reply:
I’m fine with either, but either way is not more “clean” than the other. It can still be kept clean and have date formatting. If we’re not sure about enforcing date format we can always check with design.